config/config.go

```diff
@@ -230,6 +231,26 @@ func IsProjectDir(path string) (bool, error) {
 	return fileutil.Exists(configFile, nil)
 }
 
+// IsWithinProjectDir returns true if the path is at or within an Astro project directory
+func IsWithinProjectDir(path string) (bool, error) {
+	dbtProjectPathAbs, err := filepath.Abs(filepath.Clean(path))
```

Comment:
If this function not just for DBT, can we make the variable names generic?

Reply:
Yes, good call.